benchmark/inference/inference_benchmark.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import copy
+from timeit import default_timer
+
+import torch
+from utils import get_dataset, get_model
+
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import PNAConv
+
+supported_sets = {
+    'ogbn-mag': ['rgat', 'rgcn'],
+    'reddit': ['edge_conv', 'gat', 'gcn', 'pna_conv'],
+    'ogbn-products': ['edge_conv', 'gat', 'gcn', 'pna_conv'],
+}
+
+
+def run(args: argparse.ArgumentParser) -> None:
+
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    progress_bar = True
+
+    print('BENCHMARK STARTS')
+    for dataset_name in args.datasets:
+        print(f'Dataset: {dataset_name}')
+        dataset = get_dataset(dataset_name, args.root)
+
+        hetero = True if dataset_name == 'ogbn-mag' else False
+        mask = ('paper', None) if hetero else None
+        degree = None
+
+        data = dataset[0].to(device)
+        inputs_channels = data.x_dict['paper'].size(
+            -1) if hetero else dataset.num_features
+
+        for model_name in args.models:
+            if model_name not in supported_sets[dataset_name]:
+                print(f'Configuration of {dataset_name} + {model_name} '
+                      f'not supported. Skipping.')
+                continue
+            print(f'Evaluation bench for {model_name}:')
+
+            for batch_size in args.eval_batch_sizes:
+                if not hetero:
+                    subgraph_loader = NeighborLoader(
+                        copy.copy(data),
+                        num_neighbors=[-1],
+                        input_nodes=mask,
+                        batch_size=batch_size,
+                        shuffle=False,
+                        num_workers=args.num_workers,
+                    )
+                    subgraph_loader.data.n_id = torch.arange(data.num_nodes)
+
+                for layers in args.num_layers:
+                    if hetero:
+                        subgraph_loader = NeighborLoader(
+                            copy.copy(data),
+                            num_neighbors=[args.hetero_num_neighbors] * layers,
+                            input_nodes=mask,
+                            batch_size=batch_size,
+                            shuffle=False,
+                            num_workers=args.num_workers,
+                        )
```

Comment:
Not sure I understand the differences for `NeighborLoader` between homogeneous and heterogeneous graphs, as well as the interplay between `num_neighbors` and `num_layers`. For the purposes of this benchmark, it should be the case that `len(num_neighbors) == len(num_layers)` (for both homogeneous and heterogeneous graphs).

If we want to be able to specify the number of neighbors for each layer, we can consolidate this logic and just have `num_neighbors = args.num_neighbors`. Does that make sense?